fdbclient/ManagementAPI.actor.cpp

```diff
@@ -1403,11 +1403,12 @@ ACTOR Future<int> setDDMode( Database cx, int mode ) {
 					rd >> oldMode;
 				}
 			}
-			if (!mode) {
-				BinaryWriter wrMyOwner(Unversioned());
-				wrMyOwner << dataDistributionModeLock;
-				tr.set( moveKeysLockOwnerKey, wrMyOwner.toValue() );
-			}
+			BinaryWriter wrMyOwner(Unversioned());
```

Comment:
I understand the moveKeysLockWriteKey, but why remove the !mode check?

Reply:
It seems the check is an optimization to avoid writing the ownerKeys while enabling DD mode. I felt an optimization without big value given the rarity of how many times we do this operation - so  i thought simplification has more value than optimization here.

I think in the future I would add an assert that says that the ownerKey is still same as dataDistributionModeLock and then avoid writing the ownerKey again.

Reply:
that makes sense, I verified that the data distributor will re-take the lock after it has been re-enabled so I agree it should not change anything